tests/console/orphaned_packages_check.pm

```diff
@@ -35,14 +33,18 @@ sub compare_orphans_lists {
     # A trailing or leading whitespace introduced along the path has to be removed
     # Usually it can happen in ssh like consoles (e.g. tunnel-console)
     # Input array items will be modified inside grep
-    my @missed_orphans = map { s/^\s+|\s+$//gr } @{$args{zypper_orphans}};
+    my @detected_orphans = map { s/^\s+|\s+$//gr } @{$args{zypper_orphans}};
+    delete $args{zypper_orphans};
+    my @missed_orphans = @detected_orphans;
+
     if ($args{whitelist}) {
         # Remove duplicate packages from the whitelist
         my %wl = map { $_ => 1 } (split(',', $args{whitelist}));
-        @missed_orphans = grep { !$wl{$_} } @{$args{zypper_orphans}};
+        @missed_orphans = grep { !$wl{$_} } @detected_orphans;
```

Comment:
```suggestion
        @missed_orphans = grep { !$wl{$_} } @missed_orphans;
```

Reply:
Better filter from the var that we already assigned to.

Reply:
Updated